src/realm/object-store/shared_realm.hpp

```diff
@@ -572,6 +592,14 @@ class ClosedRealmException : public std::logic_error {
     }
 };
 
+class DeleteOnOpenRealmException : public std::logic_error {
+public:
+    DeleteOnOpenRealmException()
```

Comment:
It could be helpful to add the lock file path in the exception message, in case it helps with debugging.